src/libs/ReportUtils.ts

```diff
@@ -4048,7 +4048,8 @@ function doesTransactionThreadHaveViolations(report: OnyxEntry<Report>, transact
     if (report?.stateNum !== CONST.REPORT.STATE_NUM.OPEN && report?.stateNum !== CONST.REPORT.STATE_NUM.SUBMITTED) {
         return false;
     }
-    return TransactionUtils.hasViolation(IOUTransactionID, transactionViolations);
+
+    return TransactionUtils.hasViolation(IOUTransactionID, transactionViolations) && !isSettled(IOUReportID);
```

Comment:
Can we move this call to `isSettled` outside of this function like in the proposal? I don't think it belongs in this method

Reply:
revert pr https://github.com/Expensify/App/pull/38081 has made the last change unnecessary. I will revert it.